### PR TITLE
Adds interface for asset maintenance

### PR DESCRIPTION
**Pull request by pragyeshjain**

### CODE REVIEW

File: domain/asset_maintenance.go [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+import "time"
+
+type Maintenance struct {
+	ID          string    `db:"id"`
+	AssetsID    string    `db:"assets_id"`
+	Cost        int       `db:"cost"`
+	StartedAt   time.Time `db:"started_at"`
+	EndedAt     time.Time `db:"ended_at"`
+	Description time.Time `db:"description"`
```
Comment (vkhichar) [inline domain/asset_maintenance.go]: can the 'Description' be of type 'Time' ?

File: domain/asset_maintenance.go [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package domain
+
+import "time"
+
+type Maintenance struct {
+	ID          string    `db:"id"`
```
Comment (vkhichar) [inline domain/asset_maintenance.go]: did you check the migration files ?
whats the type of ID there??

File: handler/ping_handler.go [hunk 3]
```diff
@@ -5,7 +5,7 @@ import (
 	"net/http"
 )
 
-func PingHandler() http.HandlerFunc {
+func PingHandler() http.HandlerFunc { //
```
Comment (vkhichar) [inline handler/ping_handler.go]: is this change needed?

File: repository/assets_maintenance_repo.go [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+import (
+	"context"
+
+	"github.com/jmoiron/sqlx"
+	"github.com/vkhichar/asset-management/domain"
+)
+
+type AssetMaintenanceRepo interface {
+	InsertMaintenance(ctx context.Context, assets_id string) (*domain.Maintenance, error)
```
Comment (vkhichar) [inline repository/assets_maintenance_repo.go]: How will you be able to insert the maintenance activity by only providing id in the argument??

Also, Rename this method to : 'InsertMaintenanceActivity'

File: repository/assets_maintenance_repo.go [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+	GetMaintenanceDetail(ctx context.Context, assets_id string) (*domain.Maintenance, error)
```
Comment (vkhichar) [inline repository/assets_maintenance_repo.go]: will there be only one Maintenance activity for an asset?

File: repository/assets_maintenance_repo.go [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+	"context"
+
+	"github.com/jmoiron/sqlx"
+	"github.com/vkhichar/asset-management/contract"
+	"github.com/vkhichar/asset-management/domain"
+)
+
+type UUID [16]byte
+type AssetMaintenanceRepo interface {
+	InsertMaintenanceActivity(ctx context.Context, assets_id UUID, req contract.AssetMaintain) (*domain.Maintenance, error)
```
Comment (vkhichar) [inline repository/assets_maintenance_repo.go]: input argument should be domain.Maintenance instead of contract.AssetMaintain

File: service/assets_maintenance_service.go [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+
+import (
+	"context"
+
+	"github.com/vkhichar/asset-management/domain"
+	"github.com/vkhichar/asset-management/repository"
+)
+
+type AssetMaintenanceService interface {
+	CreateAssetMaintenance(ctx context.Context, assets_id string) (user *domain.Maintenance, err error)
```
Comment (vkhichar) [inline service/assets_maintenance_service.go]: will you be able to create asset by only providing asset_id ??????

File: go.mod [hunk 8]
```diff
@@ -3,6 +3,7 @@ module github.com/vkhichar/asset-management
 go 1.13
 
 require (
+	github.com/gorilla/mux v1.8.0 // indirect
```
Comment (ranbharerp15) [inline go.mod]: We are not using mux anywhere. Please remove unused lib.

File: domain/asset_maintenance.go [hunk 9]
```diff
@@ -0,0 +1,15 @@
+package domain
+
+import (
+	"time"
+)
+
+type UUID [16]byte
+type Maintenance struct {
```
Comment (ranbharerp15) [inline domain/asset_maintenance.go]: Rename this struct as MaintenanceActivity

File: service/assets_maintenance_service.go [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+}
+
+type assetMaintenanceService struct {
+	assetMaintainRepo repository.AssetMaintenanceRepo
+}
+
+func NewAssetMaintenance(repo repository.AssetMaintenanceRepo) AssetMaintenanceService {
```
Comment (ranbharerp15) [inline service/assets_maintenance_service.go]: Set appropriate function name.

File: domain/asset_maintenance.go [hunk 11]
```diff
@@ -0,0 +1,15 @@
+package domain
+
+import (
+	"time"
+)
+
+type UUID [16]byte
```
Comment (ranbharerp15) [inline domain/asset_maintenance.go:7]: type UUID is declared in multiple files.
For UUID generation, you can use github.com/google/uuid package

**Review by ranbharerp15 — CHANGES_REQUESTED** [review]

File: repository/assets_maintenance_repo.go [hunk 12]
```diff
@@ -0,0 +1,22 @@
...
+
+	"github.com/jmoiron/sqlx"
+	"github.com/vkhichar/asset-management/domain"
+)
+
+type AssetMaintenanceRepo interface {
+	InsertMaintenanceActivity(ctx context.Context, assets_id domain.UUID, req domain.MaintenanceActivity) (*domain.MaintenanceActivity, error)
+}
+
+type assetMaintainRepo struct {
```
Comment (ranbharerp15) [inline repository/assets_maintenance_repo.go:14]: For implementation struct like this, provide empty implementation of abstract methods from parent interface. Here, as you have not provided any implementation for function InsertMaintenanceActivity in assetMaintainRepo, the build fails for such scenario. Others working with this branch have to declare it themselves for successful build.
Make sure the build is passing before raising any PR.

File: domain/asset_maintenance.go [hunk 13]
```diff
@@ -7,7 +7,7 @@ import (
 type UUID [16]byte
 type MaintenanceActivity struct {
 	ID          int       `db:"id"`
-	AssetsID    UUID      `db:"assets_id"`
+	AssetsID    UUID      `db:"asset_id"`
```
Comment (ranbharerp15) [inline domain/asset_maintenance.go]: Rename field name as well to AssetId

File: repository/assets_maintenance_repo.go [hunk 14]
```diff
@@ -8,7 +8,7 @@ import (
 )
 
 type AssetMaintenanceRepo interface {
-	InsertMaintenanceActivity(ctx context.Context, assets_id domain.UUID, req domain.MaintenanceActivity) (*domain.MaintenanceActivity, error)
+	InsertMaintenanceActivity(ctx context.Context, asset_id domain.UUID, req domain.MaintenanceActivity) (*domain.MaintenanceActivity, error)
```
Comment (ranbharerp15) [inline repository/assets_maintenance_repo.go]: We are getting AssetId in domain.MaintenanceActivity obj. We can remove this extra parameter.

File: service/assets_maintenance_service.go [hunk 15]
```diff
@@ -8,7 +8,7 @@ import (
 )
 
 type AssetMaintenanceService interface {
-	CreateAssetMaintenance(ctx context.Context, assets_id domain.UUID, req domain.MaintenanceActivity) (user *domain.MaintenanceActivity, err error)
+	CreateAssetMaintenance(ctx context.Context, asset_id domain.UUID, req domain.MaintenanceActivity) (user *domain.MaintenanceActivity, err error)
```
Comment (ranbharerp15) [inline service/assets_maintenance_service.go]: We are getting assetId in domain.MaintenanceActivity obj. We can remove this extra parameter.

File: repository/assets_maintenance_repo.go (hunk 14 above)
Comment (ranbharerp15) [reply repository/assets_maintenance_repo.go]: Follow naming convention for variable names. e.g. assetId
